### ISSUE TITLE
No internet connectivity when using Mullvad as exit node

### ISSUE THREAD

**Issue by aud** [tailscale/tailscale#10319]
### What is the issue?

When I have a Mullvad exit node selected, I lose internet connectivity. Disabling exit nodes fix the issue and I'm able to resolve network requests again.

This is the output of `tailscale status` while the Mullvad exit-node is enabled:

```
~ $ tailscale status
100.___.___.___    elliots-macbook-air-1 elliotdohm@  macOS   -
100.___.___.___    ca-tor-wg-001.mullvad.ts.net tagged-devices         active; exit node; direct 178.249.214.2:51820, tx 61544 rx 536
100.___.___.___    redacted             elliotdohm@  linux   -
100.___.___.___    redacted             elliotdohm@  android offline
100.___.___.___    redacted             elliotdohm@  linux   active; offers exit node; direct 192.168.1.33:41641, tx 9128 rx 16840
```

Eg. ping output when Mullvad exit nodes are enabled:

```
~ $ ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
Request timeout for icmp_seq 2
Request timeout for icmp_seq 3
Request timeout for icmp_seq 4
Request timeout for icmp_seq 5
Request timeout for icmp_seq 6
Request timeout for icmp_seq 7
Request timeout for icmp_seq 8
Request timeout for icmp_seq 9
Request timeout for icmp_seq 10
Request timeout for icmp_seq 11
Request timeout for icmp_seq 12
Request timeout for icmp_seq 13
Request timeout for icmp_seq 14
Request timeout for icmp_seq 15
Request timeout for icmp_seq 16
Request timeout for icmp_seq 17
Request timeout for icmp_seq 18
Request timeout for icmp_seq 19
Request timeout for icmp_seq 20
Request timeout for icmp_seq 21
Request timeout for icmp_seq 22
Request timeout for icmp_seq 23
Request timeout for icmp_seq 24
Request timeout for icmp_seq 25
Request timeout for icmp_seq 26

...

--- 1.1.1.1 ping statistics ---
89 packets transmitted, 0 packets received, 100.0% packet loss
```

### Steps to reproduce

1. Enable Tailscale, observe network requests resolving as expected
2. Enable a Mullvad exit node (eg. in my case I chose Toronto)
3. Wait 30s or so to let things propagate
4. Notice that network requests no longer resolve

### Are there any recent changes that introduced the issue?

Recently updated to Tailscale 1.54.0. Did not have this problem on the previous version.

### OS

macOS

### OS version

Sonoma 14.0

### Tailscale version

1.54.0

### Bug report

BUG-e4f48117206c116af5374318582ed9bc4fd0fcca68c7e41ad5a7cb2feb44daf3-20231118190318Z-0aac8eb665fd528d

**Comment by DentonGentry** [comment]
Around the time of the bugreport we just see timeouts trying to reach the Mullvad exit node (10.68.x.y).
Do other Mullvad exit nodes show the same behavior?

```
2023-11-18 18:58:22.3624 +0000 UTC: open-conn-track: timeout opening (TCP 10.68.x.y:63883 => 1.1.1.1:443) to node [L4RcV]; online=wg, lastRecv=3s
2023-11-18 18:58:22.362467 +0000 UTC: open-conn-track: timeout opening (TCP 10.68.x.y:63884 => 1.0.0.1:853) to node [L4RcV]; online=wg, lastRecv=3s
2023-11-18 18:58:22.362506 +0000 UTC: open-conn-track: timeout opening (TCP 10.68.x.y:63882 => 1.1.1.1:443) to node [L4RcV]; online=wg, lastRecv=3s
```

**Comment by aud** [comment]
Yes, I'm not able to get network connectivity from _any_ Mullvad exit nodes on Mac. Tested a bunch. For eg:
```
100.___.___.___   us-qas-wg-001.mullvad.ts.net tagged-devices         active; exit node; direct 198.54.135.34:51820, tx 156452 rx 380
```

Worth noting:
1. Using another non-Mullvad exit node in my Tailnet works
2. On Android I'm running 1.52.0 and have no problems using Mullvad as an exit node

**Comment by judemille** [comment]
I have this same issue. The issue occurs on my phone as well. Have not yet tested on another network -- I'm currently behind a FortiGate that I do not control.

**Comment by judemille** [comment]
Of note, the truncated output of `tailscale netcheck`:
```

Report:
	* UDP: false
	* IPv4: (no addr found)
	* IPv6: no, but OS has support
	* MappingVariesByDestIP: 
	* HairPinning: 
	* PortMapping: 
	* CaptivePortal: false
	* Nearest DERP: Chicago
```

**Comment by judemille** [comment]
Tested: the issue, for me, is limited to that environment behind a FortiGate.

**Comment by aud** [comment]
Any ideas? @DentonGentry. Happy to provide more debugging information if necessary.

**Comment by aud** [comment]
Ok, interesting, I unregistered this device in the [Tailscale Mullvad settings page](https://login.tailscale.com/admin/settings/general/mullvad) and re-registered it, and that fixed the issue.

Possibly something went wrong when I initially registered the Mac device?

**Comment by kushfest** [comment]
I'm having the same issue. I cannot connect to anything when I activate a mullvad exit node. I tried the re-registering my devices on the tailscale Mullvad settings page, but no success. As of recently this was working fine. Happy to help debug, if I can be useful.

**Comment by judemille** [comment]
@patosullivan Of curiosity: do you experience the issue only when behind a FortiGate, or at all times?

My devices have exit-node connectivity now, but **_NOT_** when behind a FortiGate, so I'm wondering if I have a separate issue to triage.

**Comment by patosullivan** [comment]
> @patosullivan Of curiosity: do you experience the issue only when behind a FortiGate, or at all times?
> 
> My devices have exit-node connectivity now, but **_NOT_** when behind a FortiGate, so I'm wondering if I have a separate issue to triage.

No idea what a FortiGate is, so probably not related for me.

**Comment by patosullivan** [comment]
> I have the same issue on **Windows 11** with Tailscale version 1.54.1
> **No custom firewall**, regular Mullvad works fine when installed.
> 
> I've tried the following, none of which fixed things:
> - enabling local LAN access, setting a global DNS override (to multiple DNS options) , with both, and with neither
> - unregistering the device and re-registering
> - restarting tailscale between making these changes
> - choosing different mullvad exit nodes
> - waiting a longer period of time after enabling
> 
> When a mullvad exit node is enabled, all external connectivity is disabled (including to other tailscale nodes). When the exit node is removed, connectivity is restored.
> 
> #> tailscale status
> 100._._._ _ _@  windows -
> 100._._._ _ _@  android offline
> 100._._._ ca-mtr-wg-001.mullvad.ts.net tagged-devices         active; exit node; direct 146.70.198.66:51820, tx 444 rx 0
> 100._._._ _ _@  android offline
> 100._._._ _ _@  android -
> 

Sounds exactly like what I'm seeing on macOS.

**Comment by seanmorton** [comment]
This has also been an issue for me the past few days, macOS app store v.1.54.1, no Fortigate, and re-registering the my device did not fix it. Disabling the Mullvad exit node with Tailscale still enabled restores connectivity.

**Comment by case** [comment]
Is anyone else here seeing similar behavior with a Fly.io Exit Node? (instead of Mullvad)

I'm seeing the same thing: as soon as I enable the Exit Node (Mac, v1.56.1) via a Fly.io app (Linux, also v1.56.1): can't access websites anymore, `ping` to both external IP addresses and domains fails with timeouts. However, `ping` within the Tailnet still works. Tailscale DNS settings have both Google Public DNS and Cloudflare DNS enabled.

Bug report `BUG-4096c579e30140cf43042b75694f211c5272887eb4bdaa63fb74f9ab50797363-20231217155335Z-8c16a532d2423afd` if this helps.

I'm using this repo, fwiw: https://github.com/patte/fly-tailscale-exit (simple Dockerfile and `start.sh` script)

**Comment by huwmartin** [comment]
Had the same issue described the last few days. Like @patosullivan, removing this device in the [Tailscale Mullvad settings page](https://login.tailscale.com/admin/settings/general/mullvad) and adding it again did not resolve my issue.

Today I noticed though that while the Tailscale macOS app said 'Device key expires in 5 months', the [Machines settings page](https://login.tailscale.com/admin/machines) in the admin console said that the key for the device had expired '1 day ago' (curiously, I had recently tried to resolve the problem by doing 'Reauthenticate' via the macOS app in the last few days, so not sure why it had expired). That weird inconsistency prompted me to try removing the machine and adding again, then adding to Tailscale Mullvad settings again, too. That has resolved my issue.

**Comment by seanmorton** [comment]
Thanks @huwmartin this workaround fixed the problem for me.

**Comment by pahute** [comment]
I am having the same issue with 1.56.1 App store
Attempted reinstall and workaround from @huwmartin - No dice

**Comment by Squeed-lol** [comment]
I was having the same issue on windows and i resolved it by uninstalling the tailscale network adapter in device manager

**Comment by kenklatt** [comment]
I experienced all the same behavior described by the OP and others here, except on NixOS. None of the deregistering, removing and re-adding, etc, suggested above worked for me.

It was my first time using an exit-node with Tailscale (not just a Mullvad one) so I went looking for any known issues which turned up [this closed issue](https://github.com/tailscale/tailscale/issues/4432). [The solution from the comments](https://github.com/tailscale/tailscale/issues/4432#issuecomment-1112819111) fixed it for me.

Perhaps there's a similar restriction in the Mac networking stack that can be loosened.

**Comment by mfdorst** [comment]
I was also experiencing this on NixOS, and that solution worked for me as well.
TL;DR the solution for nixos is
```nix
networking.firewall.checkReversePath = "loose";
```

**Comment by judemille** [comment]
I wonder if this also has interactions with firewalld? Fairly certain I use that on my laptop.

**Comment by sersorrel** [comment]
NixOS users could instead set `services.tailscale.useRoutingFeatures = "client"` (or `"server"`/`"both"`, if you want to use the machine itself as an exit node). Same result, but a little clearer why it's there when you look at your config in six months.

**Comment by bbbenji** [comment]
> I was also experiencing this on NixOS, and that solution worked for me as well. TL;DR the solution for nixos is
> 
> ```nix
> networking.firewall.checkReversePath = "loose";
> ```

This works, but breaks Docker. As in any container is now locally unreachable.

**Comment by dalaeance** [comment]
Just wanna chime in to add that I also experienced this on Android. It's what brought me here. I fixed the issue by unregistering, then re-registering my phone in the Mullvad section. 

Note: Make sure you hit "Save" after removing and after re-adding. The charges don't auto-save. Mentioning this because I hadn't realised that until my second or third try. 

In case it ever becomes relevant: 
* Pixel 7 Pro
* Android 14
* Kernel: 5.10.189-android13-4-00012-g1217bb583cc5-ab11174560 #1 Mon Dec 4 18:59:42 UTC 2023
* Tailscale version 1.60.1-t2caffeeb4-g6f7ae79a31c

**Comment by LazerDickMcCheese** [comment]
Win11, issues still happens. I've been using a Mullvad exit node for about a month now, started having this issue today. Restarts do not fix it, unregistering then re-registering the machine doesn't fix it, and it doesn't seem to be a DNS conflict as I've tried my own DNS and with overrides. Any help would be appreciated, in the meantime I may need to cancel my subscription.

**Comment by spitfire55** [comment]
For me, I found that disabling `override DNS servers` in my Admin console, and enabling `--exit-node-allow-lan-access` is what fixed it. I was trying to do both. I also didn't read the docs correctly (https://tailscale.com/kb/1258/mullvad-exit-nodes), which says overriding DNS for Mullvad exit nodes is only required for much older versions of Tailscale. I guess newer versions just work and don't require messing with DNS settings.

**Comment by mjschwenne** [comment]
For people struggling to get this to work with docker containers (at least on NixOS), enable the needed tailscale setting, `services.tailscale.useRoutingFeatures = "client"` and then start the docker or podman container with `--network=host`. On Nix, this can be done with [virtualisation.oci-containers.containers.\<name\>.extraOptions](https://search.nixos.org/options?channel=unstable&show=virtualisation.oci-containers.containers.%3Cname%3E.extraOptions&from=0&size=50&sort=relevance&type=packages&query=oci-container).
